### PR TITLE
Fix evm migration issues & rewrite feemarket migration

### PR DESCRIPTION
**Pull request by DracoLi**
Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by drklee3 — APPROVED** [review]
looks good, nice catches on migration issues

File: x/evm/module.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func (am AppModule) RegisterServices(cfg module.Configurator) {
 
 	// This migration is a Kava specific migration that also includes the
 	// upstream migrations from 3 to 5.
```
Comment (pirtleshell) [inline x/evm/module.go:150]: is this comment still true, or does it only migrate to 3?
or are we running through consensus version 5 migrations on the upstream that only corresponds to our v3?

**Review by pirtleshell — APPROVED** [review]

File: x/evm/module.go (hunk 1 above)
Comment (drklee3) [reply x/evm/module.go:150]: We combined the upstream migrations for consensus versions 3 to 5 into a single migration which is just 3 on our fork. So our version 3 is equivalent to their version 5 (with our additions on top) if that makes sense

Comment (pirtleshell) [reply x/evm/module.go:150]: thanks for the clarification!

**Pull request merged**